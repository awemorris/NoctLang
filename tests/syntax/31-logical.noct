func main() {
    var a = 0x01;
    var b = 0x02;
    print("a = " + a);
    print("b = " + b);
    print("a & b = " + (a & b));
    print("a | b = " + (a | b));
    print("a ^ b = " + (a ^ b));
    print("a << b = " + (a << b));
    print("a >> b = " + (a >> b));

    var c = 0x10;
    print("c = " + c);

    c >>= 2;
    print("c >>= 2 : " + c);

    c <<= 3;
    print("c >>= 3 : " + c);
}
