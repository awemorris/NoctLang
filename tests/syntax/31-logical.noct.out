a = 1
b = 2
a & b = 0
a | b = 3
a ^ b = 3
a << b = 4
a >> b = 0
c = 16
c >>= 2 : 4
c >>= 3 : 32
